src/main/java/io/jenkins/plugins/audit/listeners/SaveableChangeListener.java

```diff
@@ -0,0 +1,45 @@
+package io.jenkins.plugins.audit.listeners;
+
+import hudson.Extension;
+import hudson.XmlFile;
+import hudson.model.Fingerprint;
+import hudson.model.Saveable;
+import hudson.model.listeners.SaveableListener;
+import io.jenkins.plugins.audit.event.UseCredentials;
+import org.apache.logging.log4j.audit.LogEventFactory;
+
+import java.util.Hashtable;
+
+import static io.jenkins.plugins.audit.helpers.DateTimeHelper.formatDateISO;
+
+
+@Extension
+public class SaveableChangeListener extends SaveableListener {
+    /**
+     * Fired when a saveable object is created. But for now this is customized to only log
+     * the saveable fingerprint instances.
+     *
+     * @param o the saveable object.
+     * @param file the XmlFile for this saveable object.
+     */
+    @Override
+    public void onChange(Saveable o, XmlFile file) {
+
+        if(o instanceof Fingerprint){
+            UseCredentials useCredentials = LogEventFactory.getEvent(UseCredentials.class);
+            Fingerprint fp = (Fingerprint) o;
+            useCredentials.setFileName(fp.getFileName());
+            useCredentials.setName(fp.getDisplayName());
+            useCredentials.setTimestamp(formatDateISO(fp.getTimestamp().getTime()));
+            if(fp.getUsages() != null) {
+                fp.getUsages().forEach((k, v) ->{
```

Comment:
Since you're only using the values here, you can change this to something like `fp.getUsages().values().forEach(value -> { ... });`

Reply:
Looks like spotbugs is being confusing, which may have led you to insert that extra null check above. But, doing what Matt suggests here is clearer to read and clears up the spotbugs error.